lib/item.rb

```diff
@@ -0,0 +1,46 @@
+require 'httparty'
+require "pry"
+
+class Item
+  attr_reader :nypl_source, :id
+  attr_accessor :item_type_code, :location_code
+
+  def initialize(nypl_source, id)
+    @nypl_source = nypl_source
+    @id = id
+  end
+
+  def is_research
+    get_platform_api_data
+
+    is_partner || item_type_is_research || location_is_research
+  end
+
+  private
+    def is_partner
+      nypl_source == "recap-cul" || nypl_source == "recap-pul"
+    end
+
+    def item_type_is_research
+      item_collection_type = $nypl_core.by_catalog_item_type[item_type_code]
+      return item_collection_type["collectionType"][0] == "Research"
+    end
+
+    def location_is_research
+      collection_types = $nypl_core.by_sierra_location[location_code]["collectionTypes"]
```

Comment:
In this situation, would we want to return a different status code + custom error and alert that the item's corresponding item_type_code or location_code is unknown?

Reply:
It's a good question. If an item's location is unrecognized, is it better to return `false` because we can't make a positive assertion that the item is a "research" item? Or is it better to fail the request with a `5xx` level error because our data is insufficient? I'm leaning to the latter. I think there's a good chance that a new research item could be added to a new location, for example, where we'd prefer this service to throw an error rather than make an incorrect assertion about it's research-ness. See comments elsewhere about suggestion to add a new `DataError` to identify those incidents.